src/store/actions/api.actions.js

```diff
@@ -130,156 +284,166 @@ export const fetchEndpointSchema = () => (dispatch) => {
+        .then((response) => {
+            // dispatch(saveEndpointSuccess(JSON.parse(response.config.data)));
+            dispatch(saveEndpointSuccess());
             dispatch(openResponseModal({
-                status: error.response.status,
-                statusText: error.response.statusText,
-                message: error.response.data,
+                status: response.status,
+                message: 'Successfuly saved',
+                statusText: response.statusText,
+                redirectOnClose: () => (history.push('/')),
             }));
+        })
+        .catch((error) => {
+            if (error.response) {
+                dispatch(openResponseModal({
+                    status: error.response.status,
+                    statusText: error.response.statusText,
+                    message: error.response.data,
+                }));
+
+            // The request was made and the server responded with a status code
+            // that falls out of the range of 2xx
+            // More info about error handling in Axios: https://github.com/mzabriskie/axios#handling-errors
+                // eslint-disable-next-line
+                console.error(error.response.data);
+            } else if (error.request) {
+            // The request was made but no response was received
+            // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
+            // http.ClientRequest in node.js
+                // eslint-disable-next-line
+                console.log(error.request);
+            } else {
+            // Something happened in setting up the request that triggered an Error
+                // eslint-disable-next-line
+                console.log('Error', error.message);
+            }
+        });
+};
 
-        // The request was made and the server responded with a status code
-        // that falls out of the range of 2xx
-        // More info about error handling in Axios: https://github.com/mzabriskie/axios#handling-errors
-            // eslint-disable-next-line
-            console.error(error.response.data);
-        } else if (error.request) {
-        // The request was made but no response was received
-        // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
-        // http.ClientRequest in node.js
-            // eslint-disable-next-line
-            console.log(error.request);
-        } else {
-        // Something happened in setting up the request that triggered an Error
-            // eslint-disable-next-line
-            console.log('Error', error.message);
-        }
-    });
+
+export const updateEndpoint = (pathname, api) => (dispatch) => {
+    dispatch(saveEndpointRequest());
+
+    const preparedPlugins = preparePlugins(api);
+    // substitude updated list of plugins
+    const preparedApi = R.set(R.lensPath(['plugins']), preparedPlugins, api);
+
+    return client.put(`apis${pathname}`, preparedApi)
```

Comment:
async/await too

Reply:
⏫  it's done and is waiting for you here, @rafaeljesus : [https://github.com/hellofresh/janus-dashboard/pull/39](https://github.com/hellofresh/janus-dashboard/pull/39)

Reply:
Ah perfect I will review after lunch then 👍

Reply:
ta bom!